packages/special-pages/pages/duckplayer/src/js/index.js

```diff
@@ -325,23 +330,31 @@ const Comms = {
      * @param {ImportMeta['env']} opts.env
      * @param {ImportMeta['injectName']} opts.injectName
      */
-    init: (opts) => {
-        Comms.messaging = createDuckPlayerPageMessaging(opts)
-        // eslint-disable-next-line promise/prefer-await-to-then
-        Comms.messaging.getUserValues().then((value) => {
-            if ('enabled' in value.privatePlayerMode) {
-                Setting.setState(true)
-            } else {
-                Setting.setState(false)
-            }
+    init: async (opts) => {
+        const messaging = createDuckPlayerPageMessaging(opts)
+        // try to make communication with the native side.
+        const result = await callWithRetry(() => {
```

Comment:
When implementing on macOS - I encountered a race-condition where our script in the page is calling for a request/response _before_ the special page has been marked as 'safe to receive messages' by the native side.

I looked into fixing it on their end, but it's heavily related to the fact that user-scripts are global and cannot be directly tied to a particular hostname.

So, I'm fixing it here by attempting (10 times) to make an 'initial connection' - once that's established we continue as normal.

This is a nice way of ensuring we can proceed across all platforms. So whilst it's technically a macOS only fix right now, it really helps everywhere by not giving up on first error.